cli/cmd/release_promote.go

```diff
@@ -29,14 +29,23 @@ func (r *runners) InitReleasePromote(parent *cobra.Command) {
 func (r *runners) releasePromote(cmd *cobra.Command, args []string) error {
 	// parse sequence and channel ID positional arguments
 	if len(args) != 2 {
-		return errors.New("releasese sequence and channel ID are required")
+		return errors.New("release sequence and channel ID are required")
 	}
 	seq, err := strconv.ParseInt(args[0], 10, 64)
 	if err != nil {
 		return fmt.Errorf("Failed to parse sequence argument %s", args[0])
 	}
 	chanID := args[1]
 
+	if r.appType != "ship" {
+		// try to turn chanID into an actual id if it was a channel name
+		newID, err := r.api.GetChannelByName(r.appID, r.appType, chanID, "", false)
+		if err != nil {
+			return errors.Wrapf(err, "unable to get channel ID from name")
```

Comment:
wouldn't this be hit if `chanID` is an ID?

Reply:
No, GetChannelByName works for that
```
	for _, channel := range allChannels {
		if channel.Id == name || channel.Name == name {
			matchingChannels = append(matchingChannels, &types.Channel{
				ID:              channel.Id,
				Name:            channel.Name,
				Description:     channel.Description,
				ReleaseSequence: channel.ReleaseSequence,
				ReleaseLabel:    channel.ReleaseLabel,
			})
		}
	}
```